Packs/GoogleVertexAI/Integrations/GoogleVertexAI/GoogleVertexAI.py

```diff
@@ -0,0 +1,308 @@
+
+        def PaLMModel(self, prompt: str):
+            options = {"instances": [{"messages": [{"content": prompt}]}]}
+            return self._http_request(method='POST', url_suffix=f'{PROJECT_ID}/locations/us-central1/publishers/google/models/{AI_Model}', json_data=options, headers=self.headers)
+
+        
+''' MAIN FUNCTIONS '''
+
+
+def createAuthorizationURL():
+    # The client ID and access scopes are required.
+    authorization_url = f"{AUTH_URL}/oauthchooseaccount?scope={SERVICE_SCOPES}&access_type=offline&prompt=consent&response_type=code&state=state_parameter_passthrough_value&redirect_uri={REDIRECT_URI}&client_id={CLIENT_ID}"
+    return authorization_url
+
+
+def check_access_token_validation():
+    # Access tokens expires in 1 hour, then using refresh_token from test_module() function we will request for a new refresh token
+
+    integration_context: dict = get_integration_context()
+    access_token: str = integration_context.get('access_token', '')
+    valid_until: int = integration_context.get('valid_until', int)
+    time_now = epoch_seconds()
+    if len(access_token) > 0:
+        if access_token and valid_until:
+            if time_now < valid_until:
+                # Token is still valid - did not expire yet
+                return access_token
+        access_token = refresh_access_token()
+        return access_token
+    else:
+        access_token = get_access_token()
+        return access_token
+
+
+def get_access_token():
+
+    integration_context: dict = get_integration_context()
+
+    data: dict = {
+        'code': AUTH_CODE,
+        'client_id': CLIENT_ID,
+        'client_secret': CLIENT_SECRET,
+        'redirect_uri': REDIRECT_URI,
+        'grant_type': 'authorization_code'
+    }
+
+    response: requests.Response = requests.post(
+        ACCESS_TOKEN_URL,
+        data=data,
+        verify=DISABLE_SSL
+    )
+
+    if not response.ok:
+        error = error_parser(response)
+        raise ValueError(f'Failed to get access token [{response.status_code}] - {error}')
+
+    response_json: dict = response.json()
+    access_token = response_json.get('access_token', '')
+    expires_in: int = response_json.get('expires_in', 3595)
+    refresh_token = response_json.get('refresh_token', '')
+
+    time_now: int = epoch_seconds()
+    time_buffer = 5  # seconds by which to shorten the validity period
+    if expires_in - time_buffer > 0:
+        expires_in -= time_buffer
+    integration_context['refresh_token'] = refresh_token
+    integration_context['access_token'] = access_token
+    integration_context['valid_until'] = time_now + expires_in
+    set_integration_context(integration_context)
+
+    return access_token
+
+
+def refresh_access_token():
+    # A refresh token might stop working for one of these reasons: The user has revoked your app's access. The refresh token has not been used for six months
```

Comment:
Comments are usually used to explain parts of code.
Consider using doc strings if describing methods

Reply:
@thefrieddan1 Done thank you